trace_replay/trace_replay.cc

```diff
@@ -329,23 +328,29 @@ Status Replayer::MultiThreadReplay(uint32_t threads_num) {
         replay_epoch + std::chrono::microseconds(
                            (ra->trace_entry.ts - header.ts) / fast_forward_));
     if (ra->trace_entry.type == kTraceWrite) {
-      thread_pool.Schedule(&Replayer::BGWorkWriteBatch, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkWriteBatch, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceGet) {
-      thread_pool.Schedule(&Replayer::BGWorkGet, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkGet, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceIteratorSeek) {
-      thread_pool.Schedule(&Replayer::BGWorkIterSeek, ra, nullptr, nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkIterSeek, ra.release(), nullptr,
+                           nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceIteratorSeekForPrev) {
-      thread_pool.Schedule(&Replayer::BGWorkIterSeekForPrev, ra, nullptr,
-                           nullptr);
+      thread_pool.Schedule(&Replayer::BGWorkIterSeekForPrev, ra.release(),
+                           nullptr, nullptr);
       ops++;
     } else if (ra->trace_entry.type == kTraceEnd) {
-      // Do nothing for now.
+      // Do nothing for now, free the ReplayerWorkerArg.
```

Comment:
This comment does not seem to apply any more.

Reply:
Thanks pointing out. Revised.